homeassistant/components/tradfri/base_class.py

```diff
@@ -154,10 +74,7 @@ def device_info(self) -> DeviceInfo:
             via_device=(DOMAIN, self._gateway_id),
         )
 
-    def _refresh(self, device: Device, write_ha: bool = True) -> None:
-        """Refresh the device data."""
-        # The base class _refresh cannot be used, because
-        # there are devices (group) that do not have .reachable
-        # so set _attr_available here and let the base class do the rest.
-        self._attr_available = device.reachable and self._hub_available
-        super()._refresh(device, write_ha)
+    @property
+    def available(self) -> bool:
+        """Return if entity is available."""
+        return cast(bool, self._device.reachable)
```

Comment:
The coordinator entity sets available depending on last update success in the coordinator.
```suggestion
        return cast(bool, self._device.reachable) and super().available
```